E-Commerce/E-Commerce/Presentation/Views/Home/DetailView/DetailViewController.swift

```diff
@@ -66,7 +68,17 @@ class DetailViewController: UIViewController{
         checkAddToChartState()
         showAlert(message: "Product added to cart")
     }
-
+    
+    
+    @IBAction func addToFavoriteButton(_ sender: Any) {
+        changeStarButtonBackground()
+        if(starButtonIsFilled()){
```

Comment:
Just a suggestion, you can use a single line for this block like this:

starButtonIsFilled() ? FavoriteManager.shared.saveFavorite(product: self.product) : FavoriteManager.shared.deleteFavorite(product: self.product)